OpenRA.Mods.Common/Widgets/Logic/Ingame/CommandBarLogic.cs

```diff
@@ -275,15 +275,16 @@ public override void Tick()
 		void BindButtonIcon(ButtonWidget button)
 		{
 			var icon = button.Get<ImageWidget>("ICON");
-			var hasDisabled = ChromeProvider.GetImage(icon.ImageCollection, icon.ImageName + "-disabled") != null;
-			var hasActive = ChromeProvider.GetImage(icon.ImageCollection, icon.ImageName + "-active") != null;
-			var hasActiveHover = ChromeProvider.GetImage(icon.ImageCollection, icon.ImageName + "-active-hover") != null;
-			var hasHover = ChromeProvider.GetImage(icon.ImageCollection, icon.ImageName + "-hover") != null;
-
-			icon.GetImageName = () => hasActive && button.IsHighlighted() ?
-						(hasActiveHover && Ui.MouseOverWidget == button ? icon.ImageName + "-active-hover" : icon.ImageName + "-active") :
-					hasDisabled && button.IsDisabled() ? icon.ImageName + "-disabled" :
-					hasHover && Ui.MouseOverWidget == button ? icon.ImageName + "-hover" : icon.ImageName;
+
+			icon.GetImageName = () =>
+			{
+				var baseName = button.IsHighlighted() ? icon.ImageName + "-active" : icon.ImageName;
+				var hasBaseImage = ChromeProvider.GetImage(icon.ImageCollection, baseName) != null;
```

Comment:
Querying these images 60+ times a second isn't great. We can pull some things outside `GetImageName` by e.g. having separate `hasImage` vs `hasActiveImage` and otherwise use `CachedTransform` to avoid doing work when nothing has changed.